src/MsalAuthProvider.ts

```diff
@@ -27,140 +27,220 @@ import {
   };
 
-  // a person is logged in if UserAgentApplication has a current user, if there is an idtoken in the cache, and if the token in the cache is not expired
-  private isLoggedIn = () => {
-    const cacheOptions = this.UserAgentApplication.getCurrentConfiguration().cache;
-    const cacheLocation: CacheLocation =
-      cacheOptions && cacheOptions.cacheLocation ? cacheOptions.cacheLocation : 'sessionStorage';
-    const potentialLoggedInUser = this.UserAgentApplication.getAccount();
-
-    if (potentialLoggedInUser) {
-      const idToken = this.getCacheItem(cacheLocation, IDTokenKey);
-      const oldIDToken = potentialLoggedInUser.idToken as any;
-      return idToken && !this.isTokenExpired(oldIDToken);
+  public getAccountInfo = (): IAccountInfo | null => this._accountInfo;
+
+  public getToken = async (parameters?: AuthenticationParameters): Promise<AuthResponse> => {
+    const params = parameters ? parameters : this._parameters;
+
+    try {
+      const response = await this.acquireTokenSilent(params);
+
+      this.handleTokenRefreshSuccess(response);
+      this.setAuthenticationState(AuthenticationState.Authenticated);
+
+      return response;
+    } catch (error) {
+      this.dispatchAction(AuthenticationActionCreators.acquireTokenError(error));
+
+      if (error instanceof InteractionRequiredAuthError) {
+        if (this._loginType === LoginType.Redirect) {
+          this.acquireTokenRedirect(params);
+
+          // Nothing to return, the user is redirected to the login page
+          return new Promise<AuthResponse>(resolve => resolve());
+        }
+
+        try {
+          const response = await this.acquireTokenPopup(params);
+          this.handleTokenRefreshSuccess(response);
+          this.setAuthenticationState(AuthenticationState.Authenticated);
+
+          return response;
+        } catch (error) {
+          Logger.error(error);
+
+          this.dispatchAction(AuthenticationActionCreators.loginError(error));
+          this.setAuthenticationState(AuthenticationState.Unauthenticated);
+
+          throw error;
+        }
+      } else {
+        Logger.error(error);
+
+        this.dispatchAction(AuthenticationActionCreators.loginError(error));
+        this.setAuthenticationState(AuthenticationState.Unauthenticated);
+
+        throw error;
+      }
     }
+  };
+
+  public getAuthenticationParameters = (): AuthenticationParameters => this._parameters;
 
-    return false;
+  public setAuthenticationParameters = (parameters: AuthenticationParameters): void => {
+    this._parameters = parameters;
   };
 
-  private isTokenExpired = (token: any) => {
-    if (token.exp) {
-      const expirationInMs = token.exp * 1000; // AD returns in seconds
-      if (Date.now() < expirationInMs) {
-        // id token isn't expired
-        return false;
+  public registerReduxStore = (store: Store): void => {
+    this._reduxStore = store;
+    while (this._actionQueue.length) {
```

Comment:
If no redux store is registered with the authentication provider then the redux actions were being lost. Instead, we place them in a queue and replay them when the store is registered.